### ISSUE TITLE
numpy.array gives unexpected results for differently initialized values

### ISSUE THREAD

**Issue by dichotomies** [numpy/numpy#10169]
Reproducing the issue:
```py
x = [1, 2]
y = [3, 4]
a = [numpy.int16(3), numpy.int16(4)]
z = [np.array(i, j) for i, j in zip(x, a)] # type derived implicitly
# z = [np.array(i, j) for i, j in zip(x, y)] # error occurs; correctly
print(z)
# output: [array(1, dtype=int16), array(2, dtype=int16)]
```

`np.array` interpretes *the type* of the object given as second argument if the object is initialized with something like `np.int16`, `np.float64` (instead of the number of the object).

Although the documentation states that the first object should be a list or an array, it might be difficult to find the error if the user gave two numpy arrays as first 2 arguments to `np.array`.

I think it might be better to have more safety than having the functionality of deriving the type from the second argument (I don't know when I would ever need that functionality). Or at least receive a warning that the type was derived implicitly. What do you think?

**Comment by eric-wieser** [comment]
Your description is a little hard to follow. Is this a minimal example of the problem?

```python
>>> np.array(1, dtype=10)
TypeError: data type not understood
>>> np.array(1, dype=np.int16(10))
np.array(1, dtype=int16)  # you want this to match the above
```

**Comment by dichotomies** [comment]
Yes that's right. So instead of getting `np.array(1, dtype=int16)` as a results from `np.array(1, dype=np.int16(10))`, one should also get an error like for `np.array(1, dtype=10)` or at least a warning. The distilled version describes the issue precisely.
